node/src/controllers/booksController.ts

```diff
@@ -0,0 +1,76 @@
+import 'dotenv/config';
+import { Request, Response } from 'express';
+import { pool } from '../middlewares/db';
+
+export interface BookInfo {
+  book_id : number;
+  title : string;
+  author : string;
+  is_available : boolean;
+}
+
+export const getBook = async (req: Request, res: Response): Promise<void> => {
+  const { available, author } = req.query as { available?: string; author?: string };
+
+  let availableParsed: boolean | undefined;
+  if (available !== undefined) {
+    const lower = available.toLowerCase();
+    if (lower === 'true' || lower === 'false') {
+      availableParsed = lower === 'true';
+    } else {
+      res.status(400).json({ error: "Invalid 'available' query parameter. Expected 'true' or 'false'." });
+      return;
+    }
+  }
+
+  const authorParsed: string | undefined = author;
+
+  let query = 'SELECT * FROM books';
+  const params: any[] = [];
+
+  if (availableParsed !== undefined) {
+    query += params.length > 0 ? ' AND' : ' WHERE';
+    query += ' is_available = $' + (params.length + 1);
+    params.push(availableParsed);
+  }
+
+  if (authorParsed) {
+    query += params.length > 0 ? ' AND' : ' WHERE';
+    query += " author ILIKE '%' || $" + (params.length + 1) + " || '%'";
+    params.push(authorParsed);
+  }
+
+  console.debug('Executing query:', query, params);
+
+  try {
+    const result = await pool.query(query, params);
+    const books: BookInfo[] = result.rows;
+    res.json(books);
+  } catch (error) {
+    console.error('Error fetching books:', error);
+    res.status(500).json({ error: 'Internal server error' });
+  }
+};
+
+export const getBookById = async (req: Request, res: Response): Promise<void> => {
+  const { id } = req.params as { id: string };
+  const bookId = Number(id);
+
+  if (isNaN(bookId)) {
+    res.status(400).json({ error: 'Invalid book id' });
+    return;
+  }
+
```

Comment:
```suggestion
  const { id: bookId } = req.params
```
  UUID나 그런 게 PK일 수도 있고....

Reply:
그리고 sql injnection 같은건 pg 라이브러리에서 다 적절히 처리해줌

Reply:
https://github.com/validatorjs/validator.js
이걸 써보셈